### PR TITLE
Cloudways

### PR DESCRIPTION
**Pull request by ahmedkhan847**
Added more info about Cloudways. Removed info which not valid for Cloudways.

### CODE REVIEW

**Comment by ahmedkhan847** [comment]
JSON is correct now but don't where the error is comming now.

File: profiles/cloudways.json [hunk 1]
```diff
@@ -1,129 +1,311 @@
 {
   "name": "Cloudways",
-  "revision": "2014-05-28",
+  "revision": "2016-04-20",
   "url": "http://www.cloudways.com/en/",
   "status": "production",
   "status_since": "2012-03-21",
-  "type": "SaaS-centric",
+  "type": "PaaS-centric",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: There is no PaaS-centric tpye. Please see https://github.com/stefan-kolb/paas-profiles#type for more information. In your case SaaS-centric should be alright.

File: profiles/cloudways.json [hunk 2]
```diff
@@ -1,129 +1,311 @@
...
+    "region": "New York",
+    "provider": "DigitalOcean"
+  }, {
+    "continent": "AS",
+    "country": "SG",
+    "region": "Singapore",
+    "provider": "DigitalOcean"
+  }, {
+    "continent": "EU",
+    "country": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Country is missing here. Must be `GB`

File: profiles/cloudways.json [hunk 3]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "IE",
+    "region": "Ireland",
+    "provider": "AWS"
+  }, {
+    "continent": "",
+    "country": "US",
+    "region": "Iowa",
+    "provider": "GCE"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent is missing here, must be `NA`

File: profiles/cloudways.json [hunk 4]
```diff
@@ -1,129 +1,311 @@
...
+    "region": "Miami",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
+    "country": "US",
+    "region": "New York",
+    "provider": "Vultr"
+  }, {
+    "continent": "EU",
+    "country": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Country must be `GB`

File: profiles/cloudways.json [hunk 5]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "US",
+    "region": "Chicago",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
+    "country": "US",
+    "region": "Miami",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent missing.

**Comment by stefan-kolb** [comment]
Hi, thanks for your pull request :+1:

Just see my comment on the PR, so we can merge this in. 
Most of the times there are missing values for continent or country. Both of them must be set on every infrastructure.

File: profiles/cloudways.json [hunk 6]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "DE",
+    "region": "Frankfurt",
+    "provider": "AWS"
+  }, {
+    "continent": "EU",
+    "country": "IE",
+    "region": "Ireland",
+    "provider": "AWS"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent

File: profiles/cloudways.json [hunk 7]
```diff
@@ -1,129 +1,311 @@
...
+    "region": "Iowa",
+    "provider": "GCE"
+  }, {
+    "continent": "",
+    "country": "US",
+    "region": "South Carolina",
+    "provider": "GCE"
+  }, {
+    "continent": "EU",
+    "country": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Country

File: profiles/cloudways.json [hunk 8]
```diff
@@ -1,129 +1,311 @@
...
+    "region": "",
+    "provider": "GCE"
+  }, {
+    "continent": "AS",
+    "country": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Country

File: profiles/cloudways.json [hunk 9]
```diff
@@ -1,129 +1,311 @@
...
+    "region": "",
+    "provider": "GCE"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent

File: profiles/cloudways.json [hunk 10]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "US",
+    "region": "Seattle",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent

File: profiles/cloudways.json [hunk 11]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "US",
+    "region": "Silicon Valley",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent

File: profiles/cloudways.json [hunk 12]
```diff
@@ -1,129 +1,311 @@
...
+    "country": "US",
+    "region": "Los Angeles",
+    "provider": "Vultr"
+  }, {
+    "continent": "",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent

File: profiles/cloudways.json [hunk 13]
```diff
@@ -223,63 +223,63 @@
...
+    "continent": "GB",
     "country": "US",
     "region": "South Carolina",
     "provider": "GCE"
   }, {
     "continent": "EU",
-    "country": "",
-    "region": "",
+    "country": "GB",
+    "region": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: You can ommit region if it is unknown. Region is optional.

File: profiles/cloudways.json [hunk 14]
```diff
@@ -223,63 +223,63 @@
...
     "provider": "GCE"
   }, {
     "continent": "AS",
-    "country": "",
-    "region": "",
+    "country": "GB",
+    "region": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: You can ommit region if it is unknown. Region is optional.

File: profiles/cloudways.json [hunk 15]
```diff
@@ -223,63 +223,63 @@
     "region": "Ireland",
     "provider": "AWS"
   }, {
-    "continent": "",
+    "continent": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: `GB`is not a continent :smile_cat:

File: profiles/cloudways.json [hunk 16]
```diff
@@ -9,121 +9,303 @@
...
+    "country": "DE",
+    "region": "Frankfurt",
+    "provider": "AWS"
+  }, {
+    "continent": "EU",
+    "country": "IE",
+    "region": "Ireland",
+    "provider": "AWS"
+  }, {
+    "continent": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Must be `NA`

File: profiles/cloudways.json [hunk 17]
```diff
@@ -9,121 +9,303 @@
...
+    "country": "US",
+    "region": "Iowa",
+    "provider": "GCE"
+  }, {
+    "continent": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent must be `NA`

File: profiles/cloudways.json [hunk 18]
```diff
@@ -9,121 +9,303 @@
...
+    "country": "US",
+    "region": "South Carolina",
+    "provider": "GCE"
+  }, {
+    "continent": "EU",
+    "country": "GB",
+    "region": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: The two GCE entries are not correct. Please leave region empty if it is not known or correct both entries.

File: profiles/cloudways.json [hunk 19]
```diff
@@ -9,121 +9,303 @@
...
+    "provider": "GCE"
+  }, {
+    "continent": "AS",
+    "country": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Same as above and `GB` is not in Asia (`AS`)

**Comment by ahmedkhan847** [comment]
Can Stefan region be empty?

**Comment by ahmedkhan847** [comment]
Stefan GCE ASIA Pacific country is not defined that's why it's global. Country can't be empty what to do?

**Comment by stefan-kolb** [comment]
Region can be empty, country cannot.

File: profiles/cloudways.json [hunk 20]
```diff
@@ -240,7 +240,7 @@
   }, {
     "continent": "AS",
     "country": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Gb is Great Britain, that's not correct here?!

File: profiles/cloudways.json [hunk 21]
```diff
@@ -235,12 +235,12 @@
   }, {
     "continent": "EU",
     "country": "GB",
-    "region": "GB",
+    "region": "",
     "provider": "GCE"
   }, {
     "continent": "AS",
     "country": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: GB is Great Britain that not correct for country in Asia here?!

File: profiles/cloudways.json [hunk 22]
```diff
@@ -9,121 +9,303 @@
...
+    "region": "South Carolina",
+    "provider": "GCE"
+  }, {
+    "continent": "EU",
+    "country": "GB",
+    "region": "",
+    "provider": "GCE"
+  }, {
+    "continent": "AS",
+    "country": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: GB is Great Britain that not correct for country in Asia here?!

**Comment by stefan-kolb** [comment]
https://cloud.google.com/compute/docs/zones

File: profiles/cloudways.json [hunk 23]
```diff
@@ -9,121 +9,303 @@
...
+    "region": "Sydney",
+    "provider": "AWS"
+  }, {
+    "continent": "SA",
+    "country": "BR",
+    "region": "Sao Paulo",
+    "provider": "AWS"
+  }, {
+    "continent": "AS",
+    "country": "SK",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: i think this is the wrong country code, should be KR?

File: profiles/cloudways.json [hunk 24]
```diff
@@ -9,121 +9,303 @@
...
+    "region": "Iowa",
+    "provider": "GCE"
+  }, {
+    "continent": "NA",
+    "country": "US",
+    "region": "South Carolina",
+    "provider": "GCE"
+  }, {
+    "continent": "EU",
+    "country": "GB",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Is the GCE really in GB or is it in Belgium?

File: profiles/cloudways.json [hunk 25]
```diff
@@ -9,121 +9,303 @@
...
+    "region": "",
+    "provider": "GCE"
+  }, {
+    "continent": "AS",
+    "country": "TW",
+    "region": "Pacific",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Region is `Changhua County`

**Comment by stefan-kolb** [comment]
Just a few suggestions left, I think we can emrge this then :smile: :+1:

File: profiles/cloudways.json (hunk 23 above)
Comment (ahmedkhan847) [inline profiles/cloudways.json]: NO SAO Paulo is in brazil

**Comment by ahmedkhan847** [comment]
Yup I am doing this :+1:

File: profiles/cloudways.json (hunk 23 above)
Comment (stefan-kolb) [inline profiles/cloudways.json]: This line note is for the Seoul entry...

**Comment by ahmedkhan847** [comment]
What is this error now :(

**Comment by ahmedkhan847** [comment]
DO you know where is the error now :disappointed:

File: profiles/cloudways.json [hunk 26]
```diff
@@ -9,121 +9,303 @@
...
+    "country": "DE",
+    "region": "Frankfurt",
+    "provider": "Vultr"
+  }, {
+    "continent": "AS",
+    "country": "JP",
+    "region": "Tokyo",
+    "provider": "Vultr"
+  }, {
+    "continent": "AU",
```
Comment (stefan-kolb) [inline profiles/cloudways.json]: Continent `OC` :smile: and we are good to go!

**Comment by ahmedkhan847** [comment]
Any error now :smile: ?

**Comment by stefan-kolb** [comment]
Its ok, I'll merge this in. Thanks for your contribution!

**Comment by ahmedkhan847** [comment]
Your most welcome :smile:

**Pull request merged**